deployments/notifyRewardAmount/optimism/givpower_distribute_extended_sep_2024.js

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable no-use-before-define */
+const hre = require("hardhat");
+const { sendReportEmail } = require("../../mailService/mailService");
+const { ethers } = hre;
+
+const pools = [
+    {
+        address: "0x301C739CF6bfb6B47A74878BdEB13f92F13Ae5E7",
+
+        // https://github.com/Giveth/giveth-dapps-v2/issues/4434
+        amount: "1510975",
+    }, // Garden Unipool
+];
+
+// Two decimals of precision -> 615 = 6.15
+const distro = [
+    1959, 1980, 1999, 2020, 2042,
+];
+
+const initTime = 1725386400; // Timestamp of first round in seconds: Tuesday, SEP 3, 2024 18:00:00 GMT
+
+let UnipoolTokenDistributor, currentTime, nonce;
+async function main() {
+    console.log("Trying to call notifyRewardAmount...", {
+        date: new Date().toString(),
+    });
+    currentTime = Math.floor(Date.now() / 1000);
+    const [signer, ...addrs] = await ethers.getSigners();
+    nonce = await signer.getTransactionCount();
+    UnipoolTokenDistributor = await ethers.getContractFactory(
+        "UnipoolTokenDistributor",
+    );
+    await notifyRewardAmount(pools[0]);
+}
+
+async function notifyRewardAmount(pool) {
+    const unipoolTokenDistributor = await UnipoolTokenDistributor.attach(
+        pool.address,
+    );
+    const periodFinish = await unipoolTokenDistributor.periodFinish();
+    const duration = await unipoolTokenDistributor.duration();
+
+    // 10 minutes of precision
+    if (periodFinish < currentTime + 60 * 10) {
+        const pos = Math.floor((currentTime - initTime) / duration);
+        console.log("pos:", pos);
+        if (pos < 0) return;
+        if (distro[pos] === 0) return;
```

Comment:
What is this for?! in purpose?

Reply:
> What is this for?! in purpose?

I just used the last notifyReward script and changed the params and addresses, .. , didn't change the code logic, WDYT should we delete it?

Reply:
We may have put that condition in place due to some special situation we had with optimism. If the current OP rewarding program ends around the the start of the new program, we don't need that condition.